lib/chef/knife/ec_key_export.rb

```diff
@@ -25,20 +25,41 @@ class EcKeyExport < Chef::Knife
 
       include Knife::EcKeyBase
 
-      banner "knife ec key export [PATH]"
+      banner "knife ec key export [USER_DATA_PATH] [KEY_DATA_PATH]"
 
       def run
         if config[:sql_user].nil? || config[:sql_password].nil?
           load_config_from_file!
         end
 
-        path = @name_args[0] || "key_dump.json"
-        export(path)
+        # user_data_path defaults to key_dump.json to support
+        # older knife-ec-backup exports
+        user_data_path = @name_args[0] || "key_dump.json"
+        key_data_path =  @name_args[1] || "key_table_dump.json"
+
+        export(:users, user_data_path) unless config[:skip_users_table]
+
+        begin
+          export_keys(key_data_path) unless config[:skip_keys_table]
+        rescue Sequel::DatabaseError => e
+          if e.message =~ /^PG::UndefinedTable/
+            ui.error "Keys table not found. The keys table only exists on Chef Server 12."
+            ui.error "Chef Server 11 users should use the --skip-keys-table option to avoid this error."
+            exit 1
+          else
+            raise
+          end
+        end
+      end
+
+      def export_keys(path)
+        data = db.fetch('SELECT keys_by_name.*, orgs.name AS "org_name" FROM keys_by_name LEFT JOIN orgs ON keys_by_name.org_id=orgs.id')
```

Comment:
This will export only client keys, but from the name it sounds like all is expected? (Unless we created a 'global' org record, but I don't think we did.)